src/probes.cpp

```diff
@@ -71,12 +71,28 @@ void closure_entry(dyntracer_t* dyntracer,
     // }
 
     std::string function_name = function_call->get_function_name();
+
     if (function_name.compare("assign") == 0) {
-        state.process_dynamic_calls_for_closures(
-            function_name, function_call, 1);
-    } else if (function_name.compare("with") == 0) {
-        state.process_dynamic_calls_for_closures(
-            function_name, function_call, 1);
+      Argument* arg = function_call->get_argument(0);
+
+      DenotedValue* value = arg->get_denoted_value();
+
+      SEXP expr = value->get_expression(); // returns a STRSXP
+
+      SEXP expr_symbol = strsxp_to_sym(expr);
```

Comment:
Tracing should not create R objects. This can even invoke the R GC causing arbitrary R code to get executed.